### PR TITLE
Added pre-commit install instructions

### PR DESCRIPTION
**Pull request by goruha**
## what
* Added instructions for pre-commit install to README template

## why
* CPCO-246 Implement Precommit Hooks for All Repositories

## references
* https://github.com/trussworks/terraform-aws-ecs-cluster#developer-setup

- [x] Add .pre-commit-config.yaml to `templates/`
- [x] Have @johncblandii and @mcalhoun review the `.pre-commit-config.yaml` file for recommendations
- [x] Update `modules/github/Makefile.init` to install the `.pre-commit-config.yaml` file on `github/init`

### CODE REVIEW

File: templates/README.md.gotmpl [hunk 1]
```diff
@@ -198,12 +198,36 @@ In general, PRs are welcome. We follow the typical "fork-and-pull" Git workflow.
...
- 4. **Push** your work back up to your fork
- 5. Submit a **Pull Request** so that we can review your changes
+ 3. [**Setup pre-commit**](####) to verify code style
+ 4. **Commit** changes to your own branch
+ 5. **Push** your work back up to your fork
+ 6. Submit a **Pull Request** so that we can review your changes
 
 **NOTE:** Be sure to merge the latest changes from "upstream" before making a pull request!
 
+#### Setup
```
Comment (osterman) [inline templates/README.md.gotmpl]:
```suggestion
#### Developer Setup
```

File: templates/README.md.gotmpl [hunk 2]
```diff
@@ -198,12 +198,36 @@ In general, PRs are welcome. We follow the typical "fork-and-pull" Git workflow.
...
+
+Before commit please setup [pre-commit](https://pre-commit.com/) following
```
Comment (osterman) [inline templates/README.md.gotmpl]:
```suggestion
Before commiting & pushing your changes, we recommend using [`pre-commit`](https://pre-commit.com/) hooks.
```

File: templates/README.md.gotmpl [hunk 3]
```diff
@@ -198,12 +198,36 @@ In general, PRs are welcome. We follow the typical "fork-and-pull" Git workflow.
...
+  pip install pre-commit
+  pre-commit install --install-hooks
+  ```
+
+##### Windows
+  ```
+  pip install pre-commit
+  pre-commit install --install-hooks
+  ```
+
```
Comment (osterman) [inline templates/README.md.gotmpl:253]: And a section only when `test/src/*` files are present:

### Testing

This module supports automated tests using `terratest`. There are a few ways to run them. 

#### Running Tests in Docker

For docker tests, we assume you have `docker` already installed. Then export your AWS credentials (e.g. `AWS_SECRET_KEY` and `AWS_ACCESS_KEY`) before running the following command:
 
```
make -C test/src docker/test
```

### Running Natively

The following command will run `terratests` locally on your workstation, but assumes that you have working go 1.12+ environment already setup. If you don't have that, consider using the Docker testing. Make sure you have exported your credentials to the environment (E.g `AWS_SECRET_KEY`, AWS_ACCESS_KEY_ID`) before running the commands.

`make -C test/src all`

File: templates/README.md.gotmpl [hunk 4]
```diff
@@ -198,12 +198,36 @@ In general, PRs are welcome. We follow the typical "fork-and-pull" Git workflow.
 
  1. **Fork** the repo on GitHub
  2. **Clone** the project to your own machine
- 3. **Commit** changes to your own branch
- 4. **Push** your work back up to your fork
- 5. Submit a **Pull Request** so that we can review your changes
+ 3. [**Setup pre-commit**](####) to verify code style
```
Comment (osterman) [inline templates/README.md.gotmpl]:
```suggestion
 3. [**Setup pre-commit**](####) to verify code style (optional, but recommended)
```

File: templates/README.md.gotmpl (hunk 1 above)
Comment (osterman) [reply templates/README.md.gotmpl]: Consider only including this if there's a `.pre-commit-config.yaml` file.

https://github.com/cloudposse/terraform-example-module/blob/master/.pre-commit-config.yaml

**Review by osterman — DISMISSED** [review]

File: README.md [hunk 5]
```diff
@@ -341,15 +343,46 @@ If you are interested in being a contributor and want to get involved in develop
...
+ 3. [**Setup pre-commit**](#developer-setup) to verify code style (optional, but recommended)
+ 4. **Commit** changes to your own branch
+ 5. **Push** your work back up to your fork
+ 6. Submit a **Pull Request** so that we can review your changes
+
 
 **NOTE:** Be sure to merge the latest changes from "upstream" before making a pull request!
 
 
+#### Developer Setup
```
Comment (mcalhoun) [inline README.md:356]: should we consider adding `make pre-commit/install` and `make pre-commit/init` targets that automatically run the commands below?

File: templates/.pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.3.0
+  hooks:
+    - id: check-yaml
+- repo: git://github.com/antonbabenko/pre-commit-terraform
```
Comment (mcalhoun) [inline templates/.pre-commit-config.yaml:15]: maybe consider hosting a `cloudposse/pre-commit` repo with the specific hooks that we want to run? We probably want to control the code (from a security perspective) that's going to execute automatically on each commit.

Also, see this [repo](https://github.com/gruntwork-io/pre-commit) for some useful hooks we might want to use (copy to our own repo).

**Review by mcalhoun — CHANGES_REQUESTED** [review]
Overall, it looks good. I left a few comments with suggestions.

We might also want to run the `make readme` as a hook...

File: templates/.pre-commit-config.yaml (hunk 6 above)
Comment (osterman) [reply templates/.pre-commit-config.yaml:15]: I definitely support creating a cloudposse/pre-commit repo like we with github actions

File: README.md (hunk 5 above)
Comment (Gowiem) [reply README.md:356]: @mcalhoun @goruha to revive this PR a little bit: I came here looking for this as I expected it in the build-harness. I definitely think it makes sense to add a new module/target for this as it's likely not just pre-commit that needs to be brew-installed: [If you look at the Mac install instructions in pre-commit-terraform](https://github.com/antonbabenko/pre-commit-terraform#macos) those hooks also require gawk, coreutils, and a couple others that we may or may not use in the future.

**Comment by Nuru** [comment]
@osterman @mcalhoun I strongly prefer `auto-format` GitHub actions run on PRs using our hosted, controlled environment over pushing heavyweight pre-commit actions to be installed and run on heterogeneous developer workstations. We have demonstrated that auto-format works automatically and well, and for the few situations where it does not, the same support we provide the GItHub action can be used by people to run the auto-format under Docker, manually and locally.

**Comment by Nuru** [comment]
We are going to encourage people to run all pre-commit actions inside the build-harness Docker container instead of trying to get them to install hooks locally.

Superseded and closed by #286

**Pull request closed without merging**